lib/include/ert/ecl/smspec_node.hpp

```diff
@@ -56,13 +45,176 @@ typedef enum {ECL_SMSPEC_INVALID_VAR            =  0 ,
+  private:
+
+    std::string            wgname;
+    std::string            keyword;            /* The value of the KEYWORDS vector for this elements. */
+    std::string            unit;               /* The value of the UNITS vector for this elements. */
+    int                    num;                /* The value of the NUMS vector for this elements - NB this will have the value SMSPEC_NUMS_INVALID if the smspec file does not have a NUMS   vector. */
+    std::string            lgr_name;           /* The lgr name of the current variable - will be NULL for non-lgr variables. */
+    std::array<int,3>      lgr_ijk;
+
+    /*------------------------------------------- All members below this line are *derived* quantities. */
+
+    std::string            gen_key1;           /* The main composite key, i.e. WWCT:OP3 for this element. */
+    std::string            gen_key2;           /* Some of the ijk based elements will have both a xxx:i,j,k and a xxx:num key. Some of the region_2_region elements will have both a xxx:num and a xxx:r2-r2 key. Mostly NULL. */
+    ecl_smspec_var_type    var_type;           /* The variable type */
+    std::array<int,3>      ijk;                /* The ijk coordinates (NB: OFFSET 1) corresponding to the nums value - will be NULL if not relevant. */
+    bool                   rate_variable;      /* Is this a rate variable (i.e. WOPR) or a state variable (i.e. BPR). Relevant when doing time interpolation. */
+    bool                   total_variable;     /* Is this a total variable like WOPT? */
+    bool                   historical;         /* Does the name end with 'H'? */
+    int                    params_index;       /* The index of this variable (applies to all the vectors - in particular the PARAMS vectors of the summary files *.Snnnn / *.UNSMRY ). */
+    float                  default_value;      /* Default value for this variable. */
+
+    smspec_node_type();
+
+    void set_invalid_flags();
+    void init_lgr( ecl_smspec_var_type var_type ,
+                   const char * wgname  ,
+                   const char * keyword ,
+                   const char * unit    ,
+                   const char * lgr ,
+                   const char * key_join_string ,
+                   int   lgr_i, int lgr_j , int lgr_k );
+    void common_init( ecl_smspec_var_type var_type_ , const char * keyword , const std::string& unit );
+    void set_num( const int grid_dims[3] , int num_);
+    void set_keyword( const std::string& keyword_ );
+    void set_flags();
+    void init_num( ecl_smspec_var_type var_type_);
+    void set_gen_keys( const char * key_join_string_);
+    void decode_R1R2( int * r1 , int * r2)  const;
+    void set_lgr_ijk( int lgr_i , int lgr_j , int lgr_k);
+
+
+    int cmp__(const smspec_node_type * node2) const;
+    int cmp_KEYWORD_WGNAME_NUM__(const smspec_node_type * node2) const;
+    static int cmp_KEYWORD_WGNAME_NUM( const smspec_node_type * node1, const smspec_node_type * node2) {
```

Comment:
I think these functions - `cmp_KEYWORD_xxxx` can be free functions in a anonymous namespace in the `.cpp` file - no need to have them as part of the class definition.

Reply:
Problem is: they need access to private parameters or functions.